internal/cephfs/core/volume.go

```diff
@@ -38,6 +40,7 @@ import (
 // Subvolumegroup creation and volume resize decisions are
 // taken through this additional cluster information.
 var clusterAdditionalInfo = make(map[string]*localClusterState)
+var clusterAdditionalInfoMutex sync.Mutex
```

Comment:
Early suggestion:- use sync.Map instead of map and sync.Mutex

Reply:
It'll make things more complex.

We'll need to get variable and convert to clusterState type every time and then store it back.
The race condition happens only when we are writing so current code should be good.
The upper map uses a mutex and members themselves are atomic.